### PR TITLE
chore: Clean up how dev tools are initialized

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- We used callbacks to initialize our devtools and I don't think we need to
 
## Major Changes 🔄
- Use more simple logic to conditionally enable dev tools without the use of callbacks
  - This should make things cleaner and easier follow

## How to test 🧪
- Verify dev tools still work as expected in local development (`yarn dev`)
- Verify dev tools are not shown in production builds (`yarn workspace linode-manager run build && yarn workspace linode-manager run start:ci`)

### CODE REVIEW

File: packages/manager/src/dev-tools/load.ts [hunk 1]
```diff
@@ -3,29 +3,21 @@
...
-    import('./dev-tools')
-      .then((devTools) => devTools.install(store))
-      .finally(callback);
-  } else {
-    // if we don't need the DevTools, call the callback immediately.
-    callback();
-  }
+  // use a dynamic import so the dev-tools code isn't bundled with the regular
+  // app code so we don't worry about bundle size.
+  import('./dev-tools').then((devTools) => devTools.install(store));
```
Comment (abailly-akamai) [inline packages/manager/src/dev-tools/load.ts:12]: Nice! did you verify this? Meaning may not be enabled but code is still bundled

**Review by abailly-akamai — APPROVED** [review]
👍

File: packages/manager/src/dev-tools/load.ts (hunk 1 above)
Comment (bnussman-akamai) [reply packages/manager/src/dev-tools/load.ts:12]: It looks like `'./dev-tools'` **is** bundled by Vite in both `develop` and this branch, but isn't loaded by the browser unless I explicitly enable it: `http://localhost:3000/databases?dev-tools=true`. 

Should I update the comment here? Or do we want `'./dev-tools'` to not be bundled at all?

Comment (abailly-akamai) [reply packages/manager/src/dev-tools/load.ts:12]: Ah I see - I feel like comment above could be helpful, and as long as it's not bundled in production then all good 👍

Comment (jaalah-akamai) [reply packages/manager/src/dev-tools/load.ts:12]: This should not be bundled at all in my opinion. I will explain more offline

Comment (bnussman-akamai) [reply packages/manager/src/dev-tools/load.ts:10]: I pushed changes that don't allow dev tools in production unless `REACT_APP_ENABLE_DEV_TOOLS` is explicitly set. I don't know how to make Vite to **not** bundle './dev-tools' entirely.

**Pull request merged**